walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,83 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
+
+  /** Checks if the participants of the request are also attending the given event. */
+  private static boolean sharesAttendees(Event event, MeetingRequest request) {
+    Set<String> s = event.getAttendees();
+    Collection<String> c = request.getAttendees();
+    for (String person : c) {
+      if (s.contains(person)) return true;
+    }
+    return false;
+  }
+
+  /**
+   * Returns a list of times that are compatible with a given list of events.
+   * @param events A list of events where each event contains
+   * information about when a specific group of people are meeting.
+   * @param request An object describing the meeting being requested.
+   * @return A list of meeting times where all attendees are available.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> sortedTimes = new ArrayList<TimeRange>();
+    for (Event event : events) {
+      if (sharesAttendees(event, request)) sortedTimes.add(event.getWhen());
+    }
+    Collections.sort(sortedTimes, TimeRange.ORDER_BY_START);
+
+    List<TimeRange> allAvailableBlocks = new ArrayList<TimeRange>();
+    
+    /** Return empty list for an invalid request. */
```

Comment:
So, javadoc comments are for functions and some other specific things. These are generic implementation comments which should be in the format

```// COMMENT``` or ```/* COMMENT */```.